applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_nonlinear_strain_energy_response_function.cpp

```diff
@@ -31,27 +32,26 @@ namespace Kratos
             SizeType vec_size = number_of_nodes * dimension;
             mConditionsRHS[cond_it->Id()] = ZeroVector(vec_size);
             mResponseGradient_0[cond_it->Id()] = ZeroVector(vec_size);
+            mResponseGradient_1[cond_it->Id()] = ZeroVector(vec_size);
         }
     }
 
     AdjointNonlinearStrainEnergyResponseFunction::~AdjointNonlinearStrainEnergyResponseFunction()
     {
     }
 
-    // void AdjointNonlinearStrainEnergyResponseFunction::InitializeSolutionStep()
-    // {
-    //     auto condition_pointer = rModelPart.Conditions().begin();
-    //     auto cond = *condition_pointer;
-    //     Matrix residual_gradient;
-    //     Vector scaling_factor_vector;
-    //     Vector adjoint_values;
-    //     double load_factor_ratio;
-    //     ProcessInfo r_process_info = rModelPart.GetProcessInfo();
-    //     this->CalculateFirstDerivativesGradient(cond, residual_gradient,scaling_factor_vector, r_process_info);
-    //     load_factor_ratio = scaling_factor_vector[0];
-    //     KRATOS_WATCH(load_factor_ratio)
-    //     mpModelPart->GetProcessInfo().SetValue(ADJOINT_CORRECTION_FACTOR, load_factor_ratio);
-    // }
+    void AdjointNonlinearStrainEnergyResponseFunction::FinalizeSolutionStep()
+    {
+        auto condition_pointer = mpModelPart->Conditions().begin();
+        auto cond = *condition_pointer;
+        Matrix residual_gradient;
+        Vector adjoint_values;
+        double load_factor_ratio;
+        ProcessInfo r_process_info = mpModelPart->GetProcessInfo();
+        load_factor_ratio = this->CalculateAdjointScalingFactor(cond, residual_gradient, r_process_info);
```

Comment:
`double load_factor_ratio = ` the you don need the line above

Reply:
you can pass `*condition_pointer` here, saves you one line (and the undesired copy) above

Reply:
Okay, I will change it.